course_info/tests/jasmine/controllers/PeopleControllerSpec.js

```diff
@@ -704,4 +732,157 @@ describe('Unit testing PeopleController', function() {
             expect(scope.handleAjaxError.calls.count()).toEqual(1);
         });
     });
+
+    describe('confirmRemove', function() {
+        var modalContentsPartialURL =
+            'partials/remove-course-membership-confirmation.html';
+
+        beforeEach(function() {
+            var ci = {
+                course_instance_id: $routeParams.courseInstanceId,
+                title: 'confirmRemove test',
+            };
+            courseInstances.instances[ci.course_instance_id] = ci;
+            controller = $controller('PeopleController', {$scope: scope});
+        });
+
+        it('should get the partial and stick the instance on the scope', function() {
+            scope.confirmRemove();
+            expect(scope.confirmRemoveModalInstance).not.toBeNull();
+            $httpBackend.expectGET(modalContentsPartialURL).respond(200, '');
+            $httpBackend.flush(1);
+        });
+
+        it('should call removeMembership and remove itself from the scope on close',
+           function() {
+               var membership = {};
+               spyOn(scope, 'removeMembership');
+               scope.confirmRemove();
+               $httpBackend.expectGET(modalContentsPartialURL).respond(200, '');
+               $httpBackend.flush(1);
+               scope.confirmRemoveModalInstance.close(membership);
+               // TODO - figure out why this isn't being called from within jasmine
+               //expect(scope.removeMembership).toHaveBeenCalled();
+               //expect(scope.confirmRemoveModalInstance).toBeNull();
+           }
+        );
+    });
+
+    describe('removeMembership', function() {
+        var membership = {
+            profile: {
+                name_first: 'Bob',
+                name_last: 'Dobbs',
+            },
+            role: {
+                role_id: 0,
+            },
+            user_id: 'bobdobbs',
+        };
+        var courseMembershipURL = coursePeopleURL + membership.user_id;
+
+        beforeEach(function() {
+            var ci = {
+                course_instance_id: $routeParams.courseInstanceId,
+                title: 'confirmRemove test',
+            };
+            courseInstances.instances[ci.course_instance_id] = ci;
+            controller = $controller('PeopleController', {$scope: scope});
+            scope.dtInstance = {reloadData: function(){}};
+            spyOn(scope.dtInstance, 'reloadData');
+        });
+
+        it('should handle success', function() {
+            var expectedSuccess = JSON.parse(JSON.stringify(membership));
```

Comment:
Yeah, but then you have to hang it off `this`, in order to have it visible from within the `it()` function.  Tried it, looked ugly as sin.